feat: complete svelte-5 migration

ndom91 opened the pull request:
## ☕️ Reasoning

- Run svelte auto migrate script (`pnpm dlx sv migrate svelte-5`)
- Bump svelte and related dependencies in `pnpm-workspace.yml`
- Bump pnpm corepack version in root `package.json`
- Manually cleaned up a bunch of other things

Migration Guide: https://svelte.dev/docs/svelte/v5-migration-guide

## 🧢 Changes

<!--
If this PR is related to a specific issue, uncomment this section
and link it via the following text:

## 🎫 Affected issues

Fixes: INSERT_ISSUE_NUMBER

-->

<!--
If this is a WIP PR and you have todos left, feel free to uncomment this and turn this PR into a draft, see https://github.blog/2019-02-14-introducing-draft-pull-requests/

## 📌 Todos

-->

ndom91 committed:
feat: migrate remaining apps/desktop files to svelte-5

ndom91 committed:
feat: bump svelte and vite related dependencies

ndom91 committed:
fix: svelte Snippet import

ndom91 committed:
chore: prettier

ndom91 committed:
fix: repair more state assignment svelte/legacy run usage

ndom91 committed:
fix: migrate fragments -> snippets and css

mtsgrd commented on apps/desktop/src/lib/components/AIPromptEdit/AIPromptEdit.svelte:
```diff
@@ -51,9 +57,9 @@
 			}}
 		/>
 
-		{#each $userPrompts as prompt}
+		{#each $userPrompts as _prompt, idx}
 			<Content
-				bind:prompt
+				bind:prompt={$userPrompts[idx] as UserPrompt}
```
Any idea why the migration script would rewrite it this way?

ndom91 replied:
I made this change, it's required in runes mode now apparently because you can't reassign or bind in an each block argument anymore. This is the syntax they recommended

mtsgrd commented on apps/desktop/src/lib/components/AIPromptEdit/DialogBubble.svelte:
```diff
@@ -155,8 +155,8 @@
 
-		&:not(.is-error):hover,
-		&:not(.is-error):focus-within {
+		&:not(:global(.is-error)):hover,
+		&:not(:global(.is-error)):focus-within {
```
👀

ndom91 replied:
Yeah it added a lot of these. I definitley plan to check these, not sure what they're all about yet :thinking:

mtsgrd commented on apps/desktop/src/lib/components/AIPromptSelect.svelte:
```diff
@@ -35,11 +40,15 @@
 	}
 
-	$: setAllPrompts($userPrompts);
+	run(() => {
+		setAllPrompts($userPrompts);
+	});
```
Should this be an `$effect`?

mtsgrd commented on apps/desktop/src/lib/components/BackButton.svelte:
```diff
@@ -1,6 +1,8 @@
 <script>
 	import Button from '@gitbutler/ui/Button.svelte';
 	import { goto } from '$app/navigation';
+	/** @type {{children?: import('svelte').Snippet}} */
```
Maybe remove this?

mtsgrd commented on apps/desktop/src/lib/components/Board.svelte:
```diff
@@ -83,9 +93,9 @@
+		ondrop={(e) => {
 			e.preventDefault();
-			if (!dragged) {
+			if (!dragged || !sortedBranches) {
```
Is this a migration change or a bug fix?

ndom91 replied:
Bug fix. Just an extra guard I added to avoid situations where `sortedBranches` could be `undefined`

mtsgrd commented on apps/desktop/src/lib/components/Board.svelte:
```diff
@@ -11,25 +11,33 @@
+		}
+	});
+
+	let sortedBranches = $state<VirtualBranch[]>();
+
+	$effect(() => {
+		sortedBranches = $branches?.sort((a, b) => a.order - b.order) || [];
+	});
```
Looks like it could be derived.

ndom91 replied:
This gets reassigned a little further down so unfortunately can't use `derived`

ndom91 commented on apps/desktop/src/lib/commit/CommitMessageInput.svelte:
```diff
@@ -45,14 +56,16 @@
-	$: ({ title, description } = splitMessage(commitMessage));
-	$: valid = !!title;
+	let { title, description } = $derived(splitMessage(commitMessage));
+	run(() => {
+		valid = !!title;
+	});
```
Yeah exactly, the `run`'s are [a migration safe fallback](https://svelte.dev/docs/svelte/v5-migration-guide#Migration-script-run) if it's not sure whether to make them a derived or effect. So part of what I'm doing now is going through here and checking these :+1:

ndom91 committed:
fix: migrate more svelte/legacy run uses and jsdoc uses

ndom91 committed:
fix: migrate remaining svelte/legacy run placeholders

ndom91 committed:
fix: revert css :global() 

Discard changes to apps/desktop/src/lib/branch/StackingSeriesDescription.svelte
Discard changes to apps/desktop/src/lib/branch/StackingSeriesHeader.svelte
Discard changes to apps/desktop/src/lib/commit/StackingCommitCard.svelte
Discard changes to apps/desktop/src/lib/commit/StackingUpstreamCommitsAccordion.svelte
Discard changes to apps/desktop/src/lib/components/AIPromptEdit/DialogBubble.svelte
Discard changes to apps/desktop/src/lib/components/InsertEmptyCommitAction.svelte
Discard changes to apps/desktop/src/lib/components/contextmenu/ContextMenuItem.svelte
Discard changes to apps/desktop/src/lib/dropzone/LineOverlay.svelte
Discard changes to apps/desktop/src/lib/navigation/DomainButton.svelte
Discard changes to apps/desktop/src/lib/select/OptionsGroup.svelte
fix: revert more :global css

ndom91 committed:
fix: avoid trying to save invalid input like empty string to secretStore

ndom91 committed:
fix: remaining svelte-check errors

ndom91 committed:
fix: Resizer - replace dispatch with callback props

ndom91 committed:
fix: force packages/ui vitest/browser dep to use our version of playwright (1.47)

krlvi commented:
Realistically, we should land this after we have released the tauri v2 changes

ndom91 commented:
> Realistically, we should land this after we have released the tauri v2 changes 

Yeah no rush, whenever makes the most sense 👍 

Having tauri v2 released in a GA release for a bit first sounds reasonable

ndom91 commented:
Replaced by https://github.com/gitbutlerapp/gitbutler/pull/5709
